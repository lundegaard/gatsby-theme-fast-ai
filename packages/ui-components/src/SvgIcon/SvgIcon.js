import React, { forwardRef } from 'react';
import PropTypes from 'prop-types';

const SvgIcon = forwardRef(({ size = 24, ...rest }, ref) => (
	<svg
		ref={ref}
		xmlns="http://www.w3.org/2000/svg"
		width={size}
		height={size}
		viewBox="0 0 24 24"
		fill="currentcolor"
		{...rest}
	/>
));
SvgIcon.displayName = 'SvgIcon';

SvgIcon.propTypes = {
	size: PropTypes.number,
};

export default SvgIcon;
