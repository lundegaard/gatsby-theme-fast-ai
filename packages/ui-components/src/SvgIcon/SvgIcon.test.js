import React from 'react';
import { shallow } from 'enzyme';

import SvgIcon from './SvgIcon';

describe('<SvgIcon />', () => {
	const requiredProps = {};

	it('matches snapshot when shallowly rendered', () => {
		const wrapper = shallow(<SvgIcon {...requiredProps} />);
		expect(wrapper).toMatchSnapshot();
	});
});
