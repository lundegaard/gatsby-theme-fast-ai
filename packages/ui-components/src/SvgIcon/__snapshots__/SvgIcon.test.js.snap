// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SvgIcon /> matches snapshot when shallowly rendered 1`] = `
<svg
  fill="currentcolor"
  height={24}
  viewBox="0 0 24 24"
  width={24}
  xmlns="http://www.w3.org/2000/svg"
/>
`;
